app/services/hyrax/workflow/abstract_notification.rb

```diff
@@ -29,6 +30,16 @@ def message
           "#{title} (#{work_id}) was advanced in the workflow by #{user.user_key} and is awaiting approval #{comment}"
         end
 
+        # @return [document] the document (work) the the Abstract WorkFlow is creating a notification for
```

Comment:
@jcoyne Unfortunately I do not know. The public method for Entity does not have any yard comments, so I can not be certain what it is returning in all cases.  For the current case it returns a GenericWork, but I can not see how that would always be the case.  https://github.com/projecthydra-labs/hyrax/blob/master/app/models/sipity/entity.rb#L28

Maybe you have a better idea...